Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -1157,8 +1180,18 @@ bool HackStudioWidget::any_document_is_dirty() const
 
 void HackStudioWidget::update_gml_preview()
 {
-    auto gml_content = current_editor_wrapper().filename().ends_with(".gml") ? current_editor_wrapper().editor().text() : "";
+    auto filename = current_editor_wrapper().filename();
+    auto gml_content = filename.ends_with(".gml") ? current_editor_wrapper().editor().text() : "";
     m_gml_preview_widget->load_gml(gml_content);
+
+    if (m_gml_preview_dialog)
+        m_gml_preview_dialog->load_gml(gml_content, filename);
+}
+
+void HackStudioWidget::close()
+{
+    if (m_gml_preview_dialog)
+        m_gml_preview_dialog->close();
 }
```

Comment:
This is strange, there is no such concept as "closing a widget". Can this just happen in the destructor instead?

Reply:
It should be possible, that didn't even come to my head. Thank you :)

Reply:
Destructor isn't called on ``HackStudioWidget`` from my testing. This is the cleanest solution I could think of (renaming ``close()`` to ``on_application_close()`` and using ``s_window``'s ``on_close`` handler)

![image](https://user-images.githubusercontent.com/71222289/127553294-42600d14-4d2c-47e6-ace6-a0f1e6bfbba0.png)